src/DepositManager.mo

```diff
@@ -198,21 +175,20 @@ module {
         });
         case (#err err) #consolidationError(err);
       };
-      log(p, event);
+      log(entry.key(), event);
 
       // process result
       switch (res) {
         case (#ok _) {
           totalConsolidated += consolidated;
-          credit.changePool(consolidated - credited);
-          ignore release(null);
+          data.pool += feeManager.surcharge();
```

Comment:
This isn't correct. We have use values here that were captured _before_ the await* above in l. 168. Because during the await the feeManager.surcharge() could have changed. Then we would be adding the wrong value to the pool here.

Reply:
Reminder: write a test for this scenario.